commercetools-models/src/main/java/io/sphere/sdk/products/PriceDraftBuilder.java

```diff
@@ -147,12 +147,15 @@ public static PriceDraftBuilder of(final PriceDraft template) {
     }
 
     public static PriceDraftBuilder of(final Price template) {
+        final CustomFieldsDraft customFieldsDraft = Optional.ofNullable(template.getCustom())
```

Comment:
@katmatt maybe the null-check logic (which is specific only for custom fields validation) should be in the method that's related to the customFieldsDraft only ->  [`custom()`](https://github.com/commercetools/commercetools-jvm-sdk/blob/93f960ce752b7951e81ed7bb7c3baa31c250be46/commercetools-models/src/main/java/io/sphere/sdk/products/PriceDraftBuilder.java#L75): https://github.com/commercetools/commercetools-jvm-sdk/blob/93f960ce752b7951e81ed7bb7c3baa31c250be46/commercetools-models/src/main/java/io/sphere/sdk/products/PriceDraftBuilder.java#L75

I see in this builder also Null checks are also made in their own specific methods, e.g. customerGroup, channel, etc..

Reply:
Let me see if I fully understand you:. You suggest to do it like this:
```

    public PriceDraftBuilder custom(@Nullable final CustomFields custom) {
        this.custom = Optional.ofNullable(template.getCustom())
                .map(customFields -> CustomFieldsDraftBuilder.of(customFields).build())
                .orElse(null);
        return this;
    }
```

I don't like this, because I don't want to expose an additional overridden `custom` method here. Especially since we just would introduce it for our copy method factories. And the other methods are more common conversions (from `Referenceable` to `Reference`) that are useful for more use cases. 

What do you think? I hope I didn't miss something in your comment?

Reply:
I meant 

````java
 public PriceDraftBuilder custom(@Nullable final CustomFields custom) {
        this.custom = Optional.ofNullable(custom)
                .map(customFields -> CustomFieldsDraftBuilder.of(customFields).build())
                .orElse(null);
        return this;
    }
````
but why would you need to override the `custom` method here? Why not add this logic to the already existing one?

Reply:
Because the existing method has a different parameter type:
```
public PriceDraftBuilder custom(@Nullable final CustomFieldsDraft custom) {
```

Reply:
ah ok, I got confused 😁

Reply:
then does it make sense to have it in a separate method:
````java
  private static CustomFieldsDraft copyCustom(final CustomFields template) {
    return template == null ? null : CustomFieldsDraftBuilder.of(template).build();
  }
````

like it is in `CategoryDraftBuilder`, `AssetDraftBuilder`, etc?

Reply:
I totally forgot about them and yes, that's a very good idea =) And makes our code more consistent, thanks :1st_place_medal: I will update my PR 👍